drivers/I2C.cpp

```diff
@@ -120,6 +124,82 @@ void I2C::unlock() {
     _mutex->unlock();
 }
 
+int I2C::reset() {
+    lock();
+    // Try i2c_reset() i.e. target specific reset
+    i2c_reset(&_i2c);
+
+    // SCL and SDA as input pins with pullup
+    gpio_init_inout(&_gpio_scl, _scl, PIN_INPUT, PullUp, 1);
+    gpio_init_inout(&_gpio_sda, _sda, PIN_INPUT, PullUp, 1);
+
+    // Read and verify if reset was successful
+    if (gpio_read(&_gpio_scl)) {
+        if (gpio_read(&_gpio_sda)) {
+            // Return successfuly as SDA and SCL is high
+            unlock();
```

Comment:
Reset is not required every time I2C object is created. It will be required always during bootup/reset in below case. 
1. Reset occurred while data transfer was in progress on I2C (read).
2. Slave device does not have reset pin / reset mechanism and remains in slave transmit mode. In slave transmit mode, slave holds SDA low and waits for next clock pulse to finish transfer.
3. Since master went through reset, does not know of slave transmit state and tries to send address and fails.
In production grade software, it should be done during init to make sure I2C bus is good. If we decide to always call from init, I will do some cleanup on this function and make it private.

Other option is to keep it standalone, and let the user decide to call it when they face above issue. 
There is a hacky workaround in (atmel-rf-driver) [https://github.com/ARMmbed/atmel-rf-driver/pull/54] where reset is done in constructor. I don't think that to be good option.

Options:
1. Init 
2. Standalone